### ISSUE TITLE
the question of CNN

### ISSUE THREAD

**Issue by travel-go** [OpenNMT/OpenNMT-py#501]
Hello,I want to use fully convolution to train model, I set -encoder_type cnn -decoder_type cnn,but I find it ppl cannot decline when it just trained two epochs.
Epoch  2, 17000/17636; acc:  28.45; ppl:  76.05; 10812 src tok/s; 13821 tgt tok/s;   1926 s elapsed
Epoch  2, 17050/17636; acc:  28.38; ppl:  80.34; 11152 src tok/s; 13993 tgt tok/s;   1932 s elapsed
Epoch  2, 17100/17636; acc:  29.15; ppl:  76.08; 10830 src tok/s; 13867 tgt tok/s;   1938 s elapsed
Epoch  2, 17150/17636; acc:  28.67; ppl:  75.37; 11368 src tok/s; 14387 tgt tok/s;   1944 s elapsed
Epoch  2, 17200/17636; acc:  29.28; ppl:  75.99; 10909 src tok/s; 13741 tgt tok/s;   1949 s elapsed

Epoch  1, 17450/17636; acc:  30.54; ppl:  67.45; 11335 src tok/s; 14129 tgt tok/s;   1980 s elapsed
Epoch  1, 17500/17636; acc:  30.88; ppl:  67.46; 11109 src tok/s; 14092 tgt tok/s;   1985 s elapsed
Epoch  1, 17550/17636; acc:  30.93; ppl:  68.23; 10355 src tok/s; 13190 tgt tok/s;   1991 s elapsed
Epoch  1, 17600/17636; acc:  30.20; ppl:  69.98; 11412 src tok/s; 14195 tgt tok/s;   1996 s elapsed

ppl has been rising.

Should I set other parameters like change attention model ?

**Comment by srush** [comment]
Can you post your full arguments? I think there are some other options you need.

**Comment by travel-go** [comment]
Thank you for your answer!This is my parameters.
python train.py -data /data -save_model /data/demo-model -encoder_type cnn -decoder_type cnn -gpuid 1

Here is the result after I trained 13 rounds, but I think it is still a bit low, should I change the attention model?
Epoch 13, 17350/17636; acc:  44.74; ppl:  18.85; 11330 src tok/s; 14073 tgt tok/s;   2010 s elapsed
Epoch 13, 17400/17636; acc:  45.17; ppl:  17.84; 10422 src tok/s; 13453 tgt tok/s;   2015 s elapsed
Epoch 13, 17450/17636; acc:  45.76; ppl:  17.81; 10699 src tok/s; 13589 tgt tok/s;   2021 s elapsed
Epoch 13, 17500/17636; acc:  44.14; ppl:  19.18; 11035 src tok/s; 13859 tgt tok/s;   2027 s elapsed
Epoch 13, 17550/17636; acc:  45.02; ppl:  18.50; 10399 src tok/s; 13154 tgt tok/s;   2033 s elapsed
Epoch 13, 17600/17636; acc:  44.86; ppl:  18.55; 11164 src tok/s; 13982 tgt tok/s;   2039 s elapsed
Train perplexity: 18.449
Train accuracy: 45.0206
Validation perplexity: 49.0067
Validation accuracy: 30.0036
Decaying learning rate to 0.000244141

**Comment by Epsilon-Lee** [comment]
I think the ppl is not implemented as a moving average among all batches, so like all gradient-based training, the loss and the ppl will prompt even more when convergence is hit. 

You can see, in your training log, when the ppl is going higher, the acc is going down again, but I think it is normal, since training has fluctuation. I will test the cnn recently as well, if I get new experience, I would like to come back to report. @travel-go

**Comment by Epsilon-Lee** [comment]
Wait a minute, by the way, I don't quite understand the log you pasted in the very beginning, it seems at the beginning of your training, the fluctuation is happening. Did you set a too large learning rate then?

**Comment by travel-go** [comment]
I am sorry, I recently had something to do,so I did not have time to answer.Thank you for your answers.I would like to ask  CNN has this situation( the training has fluctuation), RNN will not have?
I used his initial learning rate which may be a bit high.
However, I still think the result of this CNN is a bit low. I'm not quite sure if I should change other parameters when using CNN.
